backend/src/main/java/com/huellapositiva/application/dto/ESALRequestDto.java

```diff
@@ -5,16 +5,78 @@
 import lombok.Builder;
 import lombok.Data;
 
-import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
 
 @Data
 @Builder
 @AllArgsConstructor
 public class ESALRequestDto {
 
     @Schema(
+            description = "Name of ESAL",
             example = "Canarias para todos"
     )
-    @NotEmpty
+    @NotBlank
+    @Pattern(regexp = "^[a-zA-Z ]+$")
     private final String name;
+
+    @Schema(
+            description = "ESAL's description",
+            example = "ESAL dedicada a ayudar en la recogida de alimentos."
+
+    )
+    private final String description;
+
+    @Schema(
+            description = "ESAL's webpage",
+            example = "https://www.foo-bar.com/"
+    )
+    @Pattern(regexp = "^https?://(www.)?[-a-zA-Z0-9+&@#%=~_|]*.(com|es|net|org)/[-a-zA-Z0-9+&@#%=~_|]*")
+    private final String webpage;
+
+    @Schema(
+            description = "Whether the ESAL is registered as an Volunteering Entity by Gobierno de Canarias",
+            example = "True"
+    )
+    @NotBlank
+    private final Boolean registeredEntity;
+
+    @Schema(
+            description = "Entity type of the ESAL",
+            example = "Asociacion"
+    )
+    @NotBlank
+    @Pattern(regexp = "^(Asociacion|Fundacion|Federacion Deportiva|Colegio Profesional|Club Deportivo)$")
+    private final String entityType;
```

Comment:
Field values should be language agnostic. It would be a better option to use an enum and to internationalize it.

Asociacion|Fundacion|.. are just representation details for the frontend.

Reply:
Done